src/vision/status_quo/alan_iteratively_regresses.md

```diff
@@ -0,0 +1,113 @@
+# 😱 Status quo stories: Alan iteratively regresses performance
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+A core part of DistriData, called DDSplit, is in charge of splitting input data records into fragments that are stored on distinct servers, and then reassembling those fragments back into records in response to user queries.
+
+DDSplit was originally implemented using Java code (plus some C, interfaced via JNI). Alan thinks that Rust could provide the same quality of service while requiring less memory. He decides to try reimplementing DDSplit in Rust, atop tokio.
+
+Alan wants to copy some of the abstractions he sees in the Java code that are defined via Java interfaces. Alan sees Rust traits as the closest thing to Java interfaces. However, when he experimentally defines a trait with an `async fn`, he gets the following message from the compiler:
+
+```
+error[E0706]: functions in traits cannot be declared `async`
+ --> src/main.rs:2:5
+  |
+2 |     async fn method() { }
+  |     -----^^^^^^^^^^^^^^^^
+  |     |
+  |     `async` because of this
+  |
+  = note: `async` trait functions are not currently supported
+  = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait
+```
+
+This diagnostic leads Alan to add the [async-trait crate][] as a dependency to his project. Alan then uses the `#[async_trait]` attribute provided by that crate to be able to define `async fn` methods within traits.
+
+When Alan finishes the prototype code, he finds the prototype performance has 20% slower throughput compared to the Java version.
+
+[async-trait crate]: https://crates.io/crates/async-trait
+[async-trait transform]: https://crates.io/crates/async-trait#explanation
+
+Alan is disappointed; his experience has been that Rust code performs great, (at least once you managed to get the code to be accepted by the compiler). Alan was not expecting to suffer a 20% performance hit over the Java code.
+
+The DDSplit service is being developed on a Linux machine, so Alan is able use the `perf` tool to gather sampling-based profiling data the async/await port of DDSplit. 
+
+Looking at a [flamegraph][] for the call stacks, Alan  identified many calls into the memory allocator, and many calls to `memcpy`.
```

Comment:
Where the calls high in number or high in terms of % of run time?

Reply:
(or both)

Reply:
Good question. It was high in terms of % of run time. (I don't think it was excessively high in terms of # of calls.) I will fix.

Reply:
resolved in b52d49368bd13043841676cce1b2a350e3fb6674